satpy/readers/seviri_l1b_native.py

```diff
@@ -193,10 +193,27 @@ def __init__(self, filename, filename_info, filetype_info,
         # Available channels are known only after the header has been read
         self.header_type = get_native_header(has_archive_header(self.filename))
         self._read_header()
-        self.dask_array = da.from_array(self._get_memmap(), chunks=(CHUNK_SIZE,))
+        self._make_dask_array_with_map_blocks()
         self._read_trailer()
         self.image_boundaries = ImageBoundaries(self.header, self.trailer, self.mda)
 
+    def _make_dask_array_with_map_blocks(self):
+        """Makes the dask array using the ``da.map_blocks()`` functionality."""
+        dtype = self._get_data_dtype()
+        chunks = da.core.normalize_chunks(
+            "auto",
+            shape=(self.mda["number_of_lines"],),
```

Comment:
Doesn't this shape mean the chunks are only determined for the number of lines? This shape should match the size of the final array so that "auto" chunking can use the number of columns as well as the number of rows to determine how big each chunk should be.

One way to check would be print out the shape of `self.dask_array` and the shape of `self.dask_array.compute()` and see if they match. I would have though you should pass chunks (instead of `"auto"`) as `("auto", -1)` or `("auto", 2200)` or whatever the number of columns is as well as specifying the full shape for `shape=`.

Reply:
This file format is made of interleave channels and metadata for each line, which means that we are working, for each line, with a dtype that contains all channels and more.
So this is really a 1d array of structs with are working with, hence the 1-d aspect of this chunking.
Because of this, it is not only inconvenient to chunk along the columns, but probably also negativly impacting performance.
That's why it is done like this. But thanks for pointing this out, it gave me the opportunity to document it somehow.

Reply:
Ok so the itemsize of the header type is handling the "width" of these items then so "auto" is still accurately using the users preferred chunk memory size. Sounds good.